### DISCUSSION TITLE
3.5 inch 2.88MB (4MB unformatted) Drive?

### DISCUSSION THREAD

**Discussion by Ratteler** [keirf/greaseweazle#191] [General]
I have one of these drives. Can Greaseweazle 4, or any version, be used to make it function?
I primarily work in Linux, but I'll consider anything that works.

**Comment by keirf** [comment]
It may work, but the required bandwidth is close to Full Speed USB's capacity. For reliability I either need to revise the data protocol, or you need to use a hard-to-find Greaseweazle Lightning Plus (it has High Speed USB). But you could give it a go over Full Speed USB right now.

**Comment by Holgerlx** [comment]
Tested with a Greaseweazle V4, the bandwidth seems just about sufficient. So far I hadn't had any error messages. In this context, is support for an IMG format ibm.2880 planned/possible?

﻿

**Reply by keirf** [reply]
Yes I will add ibm.2880. Remind me when I'm back next week!

**Reply by Holgerlx** [reply]
Reminder to add ibm.2880 IMG format. :-)

**Reply by keirf** [reply]
Okay, please try the test build linked as an artifact from here: https://github.com/keirf/greaseweazle/actions/runs/2867655923

Or you can find the source on branch `ibm2880`.

If it works okay I will merge to master and it will be in next release.

**Reply by keirf** [reply]
For example, if you previously made a raw dump to scp you could try `gw convert --format=ibm.2880 my.scp my.img`

**Comment by Holger6504** [comment]
My tests so far:

Greaseweazle F1
gw read image.scp			ok
gw read --format ibm.2880 image.img 	ok
gw convert --format ibm.2880 image.scp image.img	ok
gw write image.scp			nok -> Flux Underflow
gw write --format ibm.2880 image.img	nok -> Flux Underflow
Read test with HxCFloppyEmulator software ok

Greaseweazle V4
gw read image.scp			ok
gw read --format ibm.2880 image.img 	ok
gw convert --format ibm.2880 image.scp image.img	ok
gw write image.scp			ok
gw write --format ibm.2880 image.img	ok
Read test with HxCFloppyEmulator  software ok
Read test with IBM PC		ok 

Works flawlessly.
Many Thanks!

**Reply by keirf** [reply]
Could you send me or upload an scp dump of an original ED disk (not one made by greaseweazle)? Interested to see what offset track data starts from index mark. Whether it is same number of bits as an HD track or twice as many (since bits are half the size).

**Comment by Holger6504** [comment]
3,5" ED floppy image, disk formatted with Win98SE. 
[3.5_ED_Floppy.zip](https://github.com/keirf/greaseweazle/files/9365512/3.5_ED_Floppy.zip)
